_targets/build.common.imxrt

```diff
@@ -0,0 +1,166 @@
+#!/bin/bash
+#
+# Common shell script's functions for imxrt
+#
+# Copyright 2021 Phoenix Systems
+# Author: Hubert Buczynski
+#
+[ "${BASH_SOURCE[0]}" -ef "$0" ] && echo "You should source this script, not execute it!" && exit 1
+
+
+# Function gets the following argument:
+# - $1 - array contains user's commands
+#
+# Function needs external variables to be defined in a project file:
+# - MAGIC_USER_SCRIPT - number which is assigned to user script
+# - NAME_USER_SCRIPT - name of the user script
+# - KERNEL_OFFS - offset on flash memory where kernel is located
+# - SIZE_PAGE
+# - PAGE_MASK
+#
+# Output variables:
+# - PROGS - names of the applications defined in the script
+b_mkscript_user() {
+	PROGS=()
+	declare -n cmds=$1
+	mkdir -p "$PREFIX_BUILD/plo/"
+
+	ksz=$((($(wc -c < "${PREFIX_PROG_STRIPPED}phoenix-${TARGET_FAMILY}-${TARGET_SUBFAMILY}.elf") + SIZE_PAGE - 1) & PAGE_MASK))
+	poffs=$((KERNEL_OFFS + ksz))
+
+	{
+		printf "%08x\n" "$MAGIC_USER_SCRIPT"
+		for cmd in "${cmds[@]}"; do
+			args=($cmd)
+			case "${args[0]}"
+			in
+				app)
+					if [ "${args[2]}" = "-x" ]; then
+						name=${args[3]%%;*}
+					else
+						name=${args[2]%%;*}
+					fi
+
+					PROGS+=("$name")
+					psz=$((($(wc -c < "${PREFIX_PROG_STRIPPED}$name") + SIZE_PAGE - 1) & PAGE_MASK))
+					printf "alias %s 0x%x 0x%x\n" "$name" "$poffs" "$psz"
+					((poffs+=psz));;
+
+				kernel)
+					printf "alias %s 0x%x 0x%x\n" "phoenix-${TARGET_FAMILY}-${TARGET_SUBFAMILY}.elf" "$KERNEL_OFFS" "$ksz";;
+			esac;
+			printf "%s\n" "$cmd"
+		done
+
+		printf "\0"
+	} > "$PREFIX_BUILD/plo/$NAME_USER_SCRIPT"
+}
+
+
+# Function needs external variables to be defined in a project file:
```

Comment:
Whoa, really nice function descriptions. What I'm missing is one-sentence description what the function actually does (I can deduct that from the function name, but maybe not always :))